lib/shared/input_dialogs.dart

```diff
@@ -527,3 +527,78 @@ void showInputDialog<T>({
     }),
   );
 }
+
+/// Shows a dialog which takes input and offers suggestions
+Future<String?> showBlockingInputDialog<T>({

```

Comment:
This seems pretty similar to `showInputDialog`. Is there any way to combine these?

Reply:
`showInputDialog` doesn't block, so it didn't work to display the dialog where you enter your username on your first oauth login. So I modified it to create `showBlockingInputDialog`. So yea, they are pretty similar.  I think `showInputDialog` is used in one other place in the code, maybe we could make that one blocking also?

Reply:
By blocking I mean `showBlockingInputDialog` will wait for the user to click `accept` before running the code that follows it. So you can await `showBlockingInputDialog` and that doesn't work with `showInputDialog`

https://github.com/thunder-app/thunder/pull/1642/files#diff-4144a1c5223a9dd35fda6e4094acae17e6f1d3f1886b50f47977d043c54f5317R192

Reply:
I will investigate and see if I can parameterize `showInputDialog`

Reply:
Ahh ok I gotcha! In that case, I think it makes sense to transition `showInputDialog` into a `await`able method. Existing code that doesn't care to `await` can just not do that and let it run in the background. Let me know what you think.

Reply:
Yea, I think that makes sense. I can do that